src/factorysetup.c

```diff
@@ -312,19 +313,8 @@ int main(void)
     screen_splash();
     common_main();
 
-    {
-        // Set to re-enter bootloader again, otherwise we are stuck with this
-        // firmware forever.
-        auto_enter_t auto_enter = {
-            .value = sectrue_u8,
-        };
-        upside_down_t upside_down = {
-            .value = false,
-        };
-        if (!memory_bootloader_set_flags(auto_enter, upside_down)) {
-            // Not much we can do here.
-        }
-    }
+    // After reset we prefer to stay in bootloader
+    auto_enter = sectrue_u32;
```

Comment:
The factory-setup could also just erase the first 8 bytes of itself when it is done, making it unbootable (and thus the bootloader would not try to boot it).